src/Native/System.IO.Compression.Native/pal_zlib.cpp

```diff
@@ -0,0 +1,184 @@
+// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+#include "pal_zlib.h"
+#include "pal_utilities.h"
+
+#include <assert.h>
+#include <zlib.h>
+
+static_assert(PAL_Z_NOFLUSH == Z_NO_FLUSH, "");
+static_assert(PAL_Z_PARTIALFLUSH == Z_PARTIAL_FLUSH, "");
+static_assert(PAL_Z_SYNCFLUSH == Z_SYNC_FLUSH, "");
+static_assert(PAL_Z_FULLFLUSH == Z_FULL_FLUSH, "");
+static_assert(PAL_Z_FINISH == Z_FINISH, "");
+static_assert(PAL_Z_BLOCK == Z_BLOCK, "");
+
+static_assert(PAL_Z_OK == Z_OK, "");
+static_assert(PAL_Z_STREAMEND == Z_STREAM_END, "");
+static_assert(PAL_Z_NEEDDICTIONARY == Z_NEED_DICT, "");
+static_assert(PAL_Z_ERRORNO == Z_ERRNO, "");
+static_assert(PAL_Z_STREAMERROR == Z_STREAM_ERROR, "");
+static_assert(PAL_Z_DATAERROR == Z_DATA_ERROR, "");
+static_assert(PAL_Z_MEMERROR == Z_MEM_ERROR, "");
+static_assert(PAL_Z_BUFERROR == Z_BUF_ERROR, "");
+static_assert(PAL_Z_VERSIONERROR == Z_VERSION_ERROR, "");
+
+static_assert(PAL_Z_NOCOMPRESSION == Z_NO_COMPRESSION, "");
+static_assert(PAL_Z_BESTSPEED == Z_BEST_SPEED, "");
+static_assert(PAL_Z_BESTCOMPRESSION == Z_BEST_COMPRESSION, "");
+static_assert(PAL_Z_DEFAULTCOMPRESSION == Z_DEFAULT_COMPRESSION, "");
+
+static_assert(PAL_Z_FILTERED == Z_FILTERED, "");
+static_assert(PAL_Z_HUFFMANONLY == Z_HUFFMAN_ONLY, "");
+static_assert(PAL_Z_RLE == Z_RLE, "");
+static_assert(PAL_Z_FIXED == Z_FIXED, "");
+static_assert(PAL_Z_DEFAULTSTRATEGY == Z_DEFAULT_STRATEGY, "");
+
+static_assert(PAL_Z_DEFLATED == Z_DEFLATED, "");
```

Comment:
Do we use all of these?  If not, we can trim away on both the managed and native sides the ones we don't.